### PR TITLE
Update Servlet Version & MajorVersion for Servlet-5.0

### LINKED ISSUE #12021: Set ServletVersion for the servlet-5.0 feature
Outstanding Questions/Issues --
How to set the ServletVersion value in Servlet Factories. Currently, servletVersion would report incorrectly for servlet 5.0. Only one factory bundle is loaded at a time and that sets the servletversion in the webcontainer.

com.ibm.ws.webcontainer.servlet.4.0.factories/bnd.bnd -- Service-Component .

We may need to create update the transformer tool to update the manifest for these new attributes?

### PR DESCRIPTION
**Pull request by volosied**
1) Fixes #12021  Transformed servlet specVersion  to 50 using transformer tool. 

2)  Fixes part of #12105   -- Updated webcontainer servlet 4 code to return 5 as the MajorVersion when the spec level is 50. 

3)  Corrected the testServletContextMajorMinorVersion test during the EE9_FEATURES run.  (Also part of #12105 )

### CODE REVIEW

File: dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/ws/webcontainer40/osgi/webapp/WebApp40.java [hunk 1]
```diff
@@ -72,6 +73,10 @@ public WebApp40(WebAppConfiguration webAppConfig,
      */
     @Override
     public int getMajorVersion() {
+        
+        if(WebContainer.getServletContainerSpecLevel() == WebContainer.SPEC_LEVEL_50){
+            return 5;
+        }
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/ws/webcontainer40/osgi/webapp/WebApp40.java:79]: Can we limit this method to one return statement?

File: dev/wlp-jakartaee-transform/rules/jakarta-servletVersion.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Change version (occurs with all references within the file)
+40=50
+# Change name attribute back to original (should not change)
+com.ibm.ws.webcontainer.v50.dd=com.ibm.ws.webcontainer.v40.dd
```
Comment (pnicolucci) [inline dev/wlp-jakartaee-transform/rules/jakarta-servletVersion.properties:4]: Can you explain why we keep the v40 attribute?

**Comment by pnicolucci** [comment]
#run-libby-bot

**Comment by volosied** [comment]
Remove  commit since it's worked on elsewhere.  Will build one more time.

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#run-libby-bot

**Pull request merged**